File: android/sdk/src/main/java/org/jitsi/meet/sdk/JitsiMeetUserInfo.java
/*
 * Copyright @ 2019-present 8x8, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.jitsi.meet.sdk;

import android.os.Bundle;

import java.net.MalformedURLException;
import java.net.URL;

/**
 * This class represents user information to be passed to {@link JitsiMeetConferenceOptions} for
 * identifying a user.
 */
public class JitsiMeetUserInfo {
    /**
     * User's display name.
     */
    private String displayName;

    /**
     * User's email address.
     */
    private String email;

    /**
     * User's avatar URL.
     */
    private URL avatar;
    private String friendAvatarURL;
    private boolean isVoiceMode;
    private boolean isHost;
    private String friendName;

    public JitsiMeetUserInfo() {}

    public JitsiMeetUserInfo(Bundle b) {
        super();

        if (b.containsKey("displayName")) {
            displayName = b.getString("displayName");
        }

        if (b.containsKey("friendName")) {
            friendName = b.getString("friendName");
        }

        if (b.containsKey("friendAvatarURL")) {
            friendAvatarURL = b.getString("friendAvatarURL");
        }

        if (b.containsKey("email")) {
            email = b.getString("email");
        }

        if (b.containsKey("isHost")) {
            isHost = b.getBoolean("isHost");
        }

        if (b.containsKey("isVoiceMode")) {
            isVoiceMode = b.getBoolean("isVoiceMode");
        }

        if (b.containsKey("avatarURL")) {
            String avatarURL = b.getString("avatarURL");
            try {
                avatar = new URL(avatarURL);
            } catch (MalformedURLException e) {
            }
        }
    }

    public String getDisplayName() {
        return displayName;
    }

    public String getFriendName() {
        return friendName;
    }

    public String getFriendAvatarURL() {
        return friendAvatarURL;
    }

    public boolean getIsHost() {
        return isHost;
    }

    public boolean getIsVoiceMode() {
        return isVoiceMode;
    }


    public void setDisplayName(String displayName) {
        this.displayName = displayName;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public URL getAvatar() {
        return avatar;
    }

    public void setAvatar(URL avatar) {
        this.avatar = avatar;
    }

    public void setFriendName(String friendName) {
        this.friendName = friendName;
    }

    public void setFriendAvatarURL(String friendAvatarURL) {
        this.friendAvatarURL = friendAvatarURL;
    }

    public void setIsHost(boolean isHost) {
        this.isHost = isHost;
    }

    public void setIsVoiceMode(boolean isVoiceMode) {
        this.isVoiceMode = isVoiceMode;
    }

    Bundle asBundle() {
        Bundle b = new Bundle();

        if (displayName != null) {
            b.putString("displayName", displayName);
        }

        if (email != null) {
            b.putString("email", email);
        }

        if (avatar != null) {
            b.putString("avatarURL", avatar.toString());
        }

        if (friendAvatarURL != null) {
            b.putString("friendAvatarURL", friendAvatarURL);
        }

        if (friendName != null) {
            b.putString("friendName", friendName);
        }

        b.putBoolean("isHost", isHost);
        b.putBoolean("isVoiceMode", isVoiceMode);

        return b;
    }
}

File: ios/sdk/src/JitsiMeetUserInfo.m
/*
 * Copyright @ 2019-present 8x8, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#import "JitsiMeetUserInfo+Private.h"

@implementation JitsiMeetUserInfo

- (instancetype)initWithDisplayName:(NSString *)displayName
                           andEmail:(NSString *)email
                           andFriendAvatarURL:(NSString *)friendAvatarURL
                            andFriendName:(NSString *)friendName
                            andIsHost:(BOOL)isHost
                            andIsVoiceMode:(BOOL)isVoiceMode
                          andAvatar:(NSURL *_Nullable) avatar {
    self = [super init];
    if (self) {
        self.displayName = displayName;
        self.friendAvatarURL = friendAvatarURL;
        self.friendName = friendName;
        self.isHost = isHost;
        self.isVoiceMode = isVoiceMode;
        self.email = email;
        self.avatar = avatar;
    }

    return self;
}

- (NSDictionary *)asDict {
    NSMutableDictionary *dict = [[NSMutableDictionary alloc] init];

    if (self.displayName != nil) {
        dict[@"displayName"] = self.displayName;
    }

    if (self.friendAvatarURL != nil) {
        dict[@"friendAvatarURL"] = self.friendAvatarURL;
    }

    if (self.friendName != nil) {
        dict[@"friendName"] = self.friendName;
    }

    if (self.email != nil) {
        dict[@"email"] = self.email;
    }

    dict[@"isHost"] = self.isHost;
    dict[@"isVoiceMode"] = self.isVoiceMode;

    if (self.avatar != nil) {
        NSString *avatarURL = [self.avatar absoluteString];
        if (avatarURL != nil) {
            dict[@"avatarURL"] = avatarURL;
        }
    }

    return dict;
}

@end
